rust/connlib/tunnel/src/client.rs

```diff
@@ -45,18 +45,22 @@ const DNS_SENTINELS_V6: &str = "fd00:2021:1111:8000:100:100:111:0/120";
 // is 30 seconds. See resolvconf(5) timeout.
 const IDS_EXPIRE: std::time::Duration = std::time::Duration::from_secs(60);
 
-impl<CB> ClientTunnel<CB>
-where
-    CB: Callbacks + 'static,
-{
+impl ClientTunnel {
     pub fn set_resources(&mut self, resources: Vec<ResourceDescription>) {
         self.role_state.set_resources(resources);
-        self.callbacks.on_update_routes(
-            self.role_state.routes().filter_map(utils::ipv4).collect(),
-            self.role_state.routes().filter_map(utils::ipv6).collect(),
-        );
-        self.callbacks
-            .on_update_resources(self.role_state.resources());
+
+        // FIXME: It would be good to add this event from _within_ `set_resources` but we don't want to emit duplicates.
```

Comment:
I was wondering about this when reviewing the other PR for changing `add_resources` and `remove_resources` to `add_resource` and `remove_resource`.

It looks like the Client ultimately just gets a couple `update_routes` and one big `update_resources` callback, so somewhere those are de-duplicated, even though internally the resources are added one by one?